### PR TITLE
Update Camel to 4.4.1

### PR DESCRIPTION
**Pull request by claudio4j**
- Update to Camel 4.4.1 There were fixes to upstream camel-4.x branch
- Update camel-k-crds to 2.3.0

### CODE REVIEW

**Comment by claudio4j** [comment]
@squakez @oscerd for review.

File: pom.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
     <parent>
         <groupId>org.apache.camel</groupId>
         <artifactId>camel-dependencies</artifactId>
-        <version>4.0.0</version>
+        <version>4.0.4</version>
```
Comment (squakez) [inline pom.xml]: I think this should be based on `4.4.1` instead. Or maybe downstream is 4.4.0, not sure.

Comment (claudio4j) [reply pom.xml]: 4.4.1 should be the right version, driven by quarkus-camel-bom in camel-k-runtime.
However there are many changes in camel-kamelets 4.4.1 that should be backported to this repo to keep sync, I will work on this to have the required changes.

**Review by squakez — COMMENTED** [review]
Seems fine but I have not enough familiarity on the project and the implications to approve.

**Review by oscerd — APPROVED** [review]

**Pull request merged**